src/lib/Controller/GameController.js

```diff
@@ -0,0 +1,178 @@
+import AbstractController from '../Abstracts/controller';
+
+class GameController extends AbstractController {
+    constructor(model, canvas, gameElements) {
+        super(model);
+        this.canvas = canvas;
+        this.snakeModel = gameElements.snakeModel;
+        this.foodModel = gameElements.foodModel;
+        this.borderModel = gameElements.borderModel;
+        this.currentDirection = [];
+        this.interval = null;
+
+        this.startInterval = this.startInterval.bind(this);
+    }
+
+    setStartingConditions(timing) {
+        clearInterval(this.interval);
+        this.model.resetScore();
+        this.currentDirection = [this.model.get('startingDirection')];
+        this.snakeModel.set('bodySegments', [...this.model.get('snakeStartingPosition')]);
+        this.startInterval(timing);
+        this.foodModel.generateFood(this.snakeModel.get('bodySegments'));
+    }
+
+    startInterval(timing) {
+        this.interval = setInterval(() => {
```

Comment:
And if that's the main render loop of a game this should be a request animation frame